distribution/src/bin/elasticsearch-env

```diff
@@ -119,24 +119,25 @@ if [[ "$ES_DISTRIBUTION_TYPE" == "docker" ]]; then
 
   declare -a es_arg_array
 
-  while IFS='=' read -r envvar_key envvar_value
-  do
-    # Elasticsearch settings need to have at least two dot separated lowercase
-    # words, e.g. `cluster.name`, or uppercased with underscore separators and
-    # prefixed with `ES_SETTING_`, e.g. `ES_SETTING_CLUSTER_NAME`. Underscores in setting names
-    # are escaped by writing them as a double-underscore e.g. "__"
-    if [[ ! -z "$envvar_value" ]]; then
-      if [[ "$envvar_key" =~ ^[a-z0-9_]+\.[a-z0-9_]+ ]]; then
-        es_opt="-E${envvar_key}=${envvar_value}"
-        es_arg_array+=("${es_opt}")
-      elif [[ "$envvar_key" =~ ^ES_SETTING(_{1,2}[A-Z]+)+$ ]]; then
-          # The long-hand sed `y` command works in any sed variant.
-          envvar_key="$(echo "$envvar_key" | sed -e 's/^ES_SETTING_//; s/_/./g ; s/\.\./_/g; y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/' )"
-          es_opt="-E${envvar_key}=${envvar_value}"
-          es_arg_array+=("${es_opt}")
-      fi
+  # Elasticsearch settings need to either:
+  # a. have at least two dot separated lower case words, e.g. `cluster.name`, or
+  while IFS='=' read -r envvar_key envvar_value; do
+    if [[ -n "$envvar_value" ]]; then
+      es_opt="-E${envvar_key}=${envvar_value}"
+      es_arg_array+=("${es_opt}")
     fi
-  done <<< "$(env)"
+  done <<< "$(env | grep -E '^[a-z0-9_]+\.[a-z0-9_]+')"
```

Comment:
Ah wait, the regex before just checked for (roughly) `^\w+\.\w+`, but actually there can be many periods in the name. Maybe we can mirror what we now have for the `ES_SETTING` variant e.g.

```suggestion
  done <<< "$(env | grep -E '^[a-z0-9_]+(\.[a-z0-9_]+)+=')"
```

Reply:
Just tested your proposal locally (note that I am only testing the variables from the tests that failed thus far), and this works as well. What is the protocal? Can/should I click the _Commit suggestion_ button?

Reply:
Yes that works, or you can push the change yourself. It's up to you really, we'll do a squash commit anyway when we merge 👍